src/Aspire.Hosting/Dashboard/proto/resource_service.proto

```diff
@@ -170,6 +170,16 @@ message Resource {
 
     // The set of volumes mounted to the resource. Only applies to containers.
     repeated Volume volumes = 15;
+
+    // The aggregate health state of the resource
+    optional HealthStateKind HealthState = 16;
```

Comment:
Since this part is public API, I think we should do the right thing here and have complete health state information. In other words, have a list of the different health checks and what there status is.

e.g.

```
repeated Health health_stats = 16;

message Health {
    string name = 1;
    HealthStateKind state = 2;
}
```

Then in the future it will be easier to add more detailed information about the health status, such as a list in the resource details.

Reply:
You want this to be computed twice?

Reply:
I don't see a problem with the dashboard calculating it.

Reply:
I think we want a single source of truth. The resource server in the case of the apphost needs the aggregate as well (its how WaitFor works), so it makes sense to avoid recomputing it unnecessarily. I think we should send the aggregate and defer the health check detail for when we invent UI to show them.